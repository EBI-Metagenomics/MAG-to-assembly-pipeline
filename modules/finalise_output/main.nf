process FINALISE_OUTPUT {
    label "python_based"
    conda "${moduleDir}/environment.yml"
    
    input:
    path input
    path not_linked_mags
    path catalogue_metadata
    path previous_processed_acc, stageAs: 'previous_processed_acc.tsv'
    path previous_table, stageAs: 'previous_table.tsv'


    output:
    path 'mag_to_assembly_links_*.tsv'
    path 'processed_accessions_*.tsv'

    script:
    def processed_accessions = previous_processed_acc ? "previous_processed_acc.tsv" : ""
    def previous_linking_table = previous_table ? "--previous-table previous_table.tsv" : ""
    """

    cat ${input} ${not_linked_mags} ${processed_accessions} | cut -f 1 > processed_accessions_\$(date +"%Y-%m-%d_%Hh%Mm").tsv

    finalise_output.py \
        ${previous_linking_table} \
        --catalogue-metadata ${catalogue_metadata} \
        ${input}

    """
}