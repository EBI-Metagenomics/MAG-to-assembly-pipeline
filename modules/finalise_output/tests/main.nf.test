nextflow_process {

    name "Test Process FINALISE_OUTPUT"
    script "../main.nf"
    process "FINALISE_OUTPUT"

    test("Output files preparation") {

        when {
            process {
                """
                input[0] = Channel.fromPath("${moduleDir}/tests/data/*links.tsv", checkIfExists: true).collect()
                input[1] = Channel.fromPath("${moduleDir}/tests/data/*not_linked.tsv", checkIfExists: true).collect()
                input[2] = file("${moduleDir}/tests/data/all_catalog_metadata.tsv")
                input[3] = file("${projectDir}/data/tests/processed_accessions.tsv")
                input[4] = file("${projectDir}/data/tests/mag_to_assembly_links.tsv")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
