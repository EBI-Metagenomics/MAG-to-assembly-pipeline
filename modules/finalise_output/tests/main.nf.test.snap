{
    "Output files preparation": {
        "content": [
            {
                "0": [
                    "mag_to_assembly_links_2024-08-15_15h00m.tsv:md5,083454f116f953fd5569a253c168aaf7"
                ],
                "1": [
                    "processed_accessions_2024-08-15_15h00m.tsv:md5,6a007e373a51fb05ed4dfdcdf8bd9e85"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "23.04.3"
        },
        "timestamp": "2024-08-15T15:01:23.288577431"
    }
}